src/preferArrayLiteralRule.ts

```diff
@@ -6,16 +6,44 @@ import { AstUtils } from './utils/AstUtils';
 import { ExtendedMetadata } from './utils/ExtendedMetadata';
 import { isObject } from './utils/TypeGuard';
 
+// undefined for case when function/constructor is called directly without namespace
+const RESTRICTED_NAMESPACES = [undefined, 'window', 'global', 'globalThis'];
```

Comment:
Nit: just for semantic correctness, how about a `Set`?

Reply:
Yeah, sure. Always remember that Node 6 doesn't support `includes` but forgot about `Set`. Will update implementation